snuba/query/processors/uuid_column_processor.py

```diff
@@ -0,0 +1,112 @@
+from typing import Set
+import uuid
+
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.conditions import (
+    binary_condition,
+    is_condition,
+    is_in_condition,
+    is_not_in_condition,
+)
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class UUIDColumnProcessor(QueryProcessor):
+    """
+    If a condition is being performed on a column that stores UUIDs (as defined in the constructor)
+    then change the condition to use a proper UUID instead of a string.
+    """
+
+    def __init__(self, uuid_columns: Set[str]) -> None:
+        self.__uuid_columns = uuid_columns
+
+    def is_uuid_column(self, exp: Expression) -> bool:
+        return isinstance(exp, Column) and exp.column_name in self.__uuid_columns
+
+    def is_formatted_uuid(self, exp: Expression) -> bool:
+        return (
+            isinstance(exp, FunctionCall)
+            and exp.function_name == "replaceAll"
+            and isinstance(exp.parameters[0], FunctionCall)
+            and exp.parameters[0].function_name == "toString"
+            and isinstance(exp.parameters[0].parameters[0], Column)
+            and exp.parameters[0].parameters[0].column_name in self.__uuid_columns
+        )
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_condition(exp: Expression) -> Expression:
+            if is_condition(exp):
+                assert isinstance(exp, FunctionCall)
+                if len(exp.parameters) > 1:
+                    # This doesn't seem great but here it is: There is a separate processor that converts
```

Comment:
I assume you are not making the change to the logic of the processor that introduces the replaceAll to begin with (to avoid doing so in conditions) so that you can apply the same logic to promoted contexts as well .... well .... when we start promoting promoted contexts. Is this the reason?

Reply:
Yeah, I thought about removing the code that converts event_id to use replaceAll etc but it seemed a lot more involved than undoing the work. And having this logic like this makes it a little more portable.